src/devices/tuya.ts

```diff
@@ -10651,6 +10651,120 @@ const definitions: Definition[] = [
+                .withUnit('m')
+                .withDescription('Target distance'),
+            e.illuminance_lux(),
+            e.binary('indicator', ea.STATE_SET, 'ON', 'OFF').withDescription('LED Indicator'),
+            e
+                .numeric('none_delay_time', ea.STATE_SET)
+                .withValueMin(0).withValueMax(28800)
+                .withValueStep(1).withUnit('Sec')
+                .withDescription('Hold delay time'),
+            e
+                .numeric('move_detection_max', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Move detection max distance'),
+            e
+                .numeric('move_detection_min', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Move detection min distance'),
+            e
+                .numeric('small_move_detection_max', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(6)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Small move detection max distance'),
+            e
+                .numeric('small_move_detection_min', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(6)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Small move detection min distance'),
+            e
+                .numeric('breath_detection_max', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(6)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Breath detection max distance'),
+            e
+                .numeric('breath_detection_min', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(6)
+                .withValueStep(0.01)
+                .withUnit('m')
+                .withDescription('Breath detection min distance'),
+            e
+                .numeric('move_sensitivity', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(1)
+                .withDescription('Move sensitivity'),
+            e
+                .numeric('breath_sensitivity', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(1)
+                .withDescription('Breath sensitivity'),
+            e
+                .numeric('small_move_sensitivity', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(1)
+                .withDescription('Small Move sensitivity'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [
```

Comment:
Could you also return a `presence` here (a boolean) and add `e.presence()` to `exposes`? Example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/0f0917687d962efcca7788ac392d32fe76afdc16/src/devices/onenuo.ts#L30

Reply:
Done and tested :)
Now I have to figure out how to get it in here haha. Let's try this
```suggestion
                [
                    1,
                    null,
                    {
                        from: (v) => {
                            const lookup = {none: tuya.enum(0), presence: tuya.enum(1), peaceful: tuya.enum(2), 'small movement': tuya.enum(3), 'large movement': tuya.enum(4)};
                            const presenceState = Object.entries(lookup).find((i) => i[1].valueOf() === v)[0];
                            return {
                                presence: presenceState != 'none',
                                presence_state: presenceState,
                            };
                        },
                    },
                ],
```